Framework/DataHandling/inc/MantidDataHandling/MeshFileIO.h

```diff
@@ -0,0 +1,100 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "MantidGeometry/Objects/MeshObject.h"
+#include "MantidKernel/Logger.h"
+#include "MantidKernel/Matrix.h"
+#include "MantidKernel/V3D.h"
+
+#ifndef MANTID_DATAHANDLING_MESHFILEIO_H_
+#define MANTID_DATAHANDLING_MESHFILEIO_H_
+namespace {
+Mantid::Kernel::Logger g_log("MeshFileIO");
+}
+namespace Mantid {
+namespace DataHandling {
+enum class ScaleUnits { undefined, metres, centimetres, millimetres };
+
+class DLLExport MeshFileIO {
+public:
+  Kernel::Matrix<double> generateMatrix(double xRotation, double yRotation,
+                                        double zRotation);
+
+  boost::shared_ptr<Geometry::MeshObject>
+  rotate(boost::shared_ptr<Geometry::MeshObject> environmentMesh,
+         double xRotation, double yRotation, double zRotation);
+
+  Kernel::Matrix<double> generateXRotation(double xRotation);
+  Kernel::Matrix<double> generateYRotation(double yRotation);
+  Kernel::Matrix<double> generateZRotation(double zRotation);
+
+  boost::shared_ptr<Geometry::MeshObject>
+  translate(boost::shared_ptr<Geometry::MeshObject> environmentMesh,
+            const std::vector<double> translationVector);
+
+  Kernel::V3D createScaledV3D(double xVal, double yVal, double zVal);
+
+protected:
+  MeshFileIO(ScaleUnits scaleType) : m_scaleType(scaleType) {}
+  MeshFileIO(ScaleUnits scaleType, std::vector<uint32_t> triangles,
+             std::vector<Kernel::V3D> vertices)
+      : m_scaleType(scaleType), m_triangle(triangles), m_vertices(vertices) {}
+  double scaleValue(double val) {
+    switch (m_scaleType) {
+    case ScaleUnits::centimetres:
+      return val / 100;
+    case ScaleUnits::millimetres:
+      return val / 1000;
+    case ScaleUnits::metres:
+      return val;
+    default:
+      return val;
+    }
+  }
+
+  float removeScale(double value) {
+    switch (m_scaleType) {
+    case ScaleUnits::centimetres:
+      value = value * 100;
+      break;
+    case ScaleUnits::millimetres:
+      value = value * 1000;
+      break;
+    case ScaleUnits::metres:
+      break;
+    default:; // do nothing
+    }
+    return float(value);
+  };
+
+  ScaleUnits m_scaleType;
+  std::vector<uint32_t> m_triangle;
+  std::vector<Kernel::V3D> m_vertices;
+
+  inline void setScaleType(const ScaleUnits scaleType) {
```

Comment:
The inline specifier against setScaleType is redundant because any class member functions defined in the memberlist are implicitly inline (so scaleValue and removeScale are also implicitly inline). I'll remove the inline specifier from setScaleType. 
The intention is that all the functions in this header are inline because they're small and having them inline is good for performance